### PR TITLE
Refactor test suite

### PR DESCRIPTION
**Pull request by Pav0l**
# Description

Refactoring test suite to make tests more independent from each other

## Checklist

Remove any items which are not applicable.

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works AND the tests pass

### CODE REVIEW

**Comment by Pav0l** [comment]
[Refactor BE test suite to have independent tests](https://trello.com/c/CpB1fWUS/133-refactor-be-test-suite-to-have-independent-tests)

File: tests/farmers.test.js [hunk 1]
```diff
@@ -193,69 +197,61 @@ describe('Farmer route', () => {
...
         res.should.have.status(403);
         res.body.message.should.equal(
           'This Farmer is archived and can not be updated'
         );
-        done(err);
       });
   });
 
-  it('It should return 404 if there are is no famrme with the id in the db', done => {
+  it('It should return 404 if there are no farmers in the DB', async () => {
```
Comment (quansenB) [inline tests/farmers.test.js]: This is not a good name for this test as it checks for only one farmerById

File: tests/farmerInput.js [hunk 2]
```diff
@@ -18,7 +18,7 @@ const farmerInput = {
     house_number: '1',
     nearest_landmark: 'Bustop',
     Phone_1: '8765456789',
-    Phone_2: '678976546', 
+    Phone_2: '678976546',
```
Comment (quansenB) [inline tests/farmerInput.js]: what a catch :D

File: tests/users.test.js [hunk 3]
```diff
@@ -71,46 +71,44 @@ describe('Users route [/user/login]', () => {
...
-      .send(seeds.missingUsername)
+      .send(seeds.missingUsernameLogin)
       .end((err, res) => {
         res.should.have.status(400);
         done(err);
       });
   });
 
-  it('should return 401 when user does not exist', (done) => {
+  it('Should return 401 when user does not exist', (done) => {
```
Comment (quansenB) [inline tests/users.test.js]: This test is very confusing, what does it do? Why does it use the signup route?

Comment (quansenB) [reply tests/users.test.js]: oh it doesn't provide a token, so it fails. Please rename the description

File: tests/testsSetup.js [hunk 4]
```diff
@@ -16,51 +19,77 @@ const staffUser = {
...
   } catch (error) {
-    console.log(error);
+    console.error(error.name, error.message);
   }
 });
 
 after(async () => {
-  try {
-    await connectDB();
+  await models.User.deleteMany({});
```
[the remark this answers is not in the document]
Comment (Pav0l) [reply tests/testsSetup.js]: The function is called in `try-catch` blocks, so no need to have it in function declaration as well.

File: tests/farmerInput.js (hunk 2 above)
Comment (Pav0l) [reply tests/farmerInput.js]: auto-format 🔥

**Pull request merged**